packages/expo-file-system/android/src/main/java/expo/modules/filesystem/FilePermissionChecker.java

```diff
@@ -0,0 +1,55 @@
+package expo.modules.filesystem;
+
+import android.content.Context;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.EnumSet;
+import java.util.List;
+
+import expo.core.interfaces.InternalModule;
+import expo.interfaces.filesystem.FilePermissionWielderInterface;
+import expo.interfaces.filesystem.Permission;
+
+public class FilePermissionChecker implements FilePermissionWielderInterface, InternalModule {
+
+  @Override
+  public List<Class> getExportedInterfaces() {
+    return Collections.<Class>singletonList(FilePermissionWielderInterface.class);
+  }
+
+  @Override
+  public EnumSet<Permission> getInfo(Context context, String path) {
+    try {
```

Comment:
Frankly speaking I don't feel comfortable mixing logic like this. It's neither here nor there. 😕 How about making `getInfo` (or `getPathPermissions` after rename) call other two functions:?
- `getInternalPathPermissions(Context context, String internalPath)`
- `getExternalPathPermissions(String externalPath)`

What do you think about it?